model_managers/model_manager.py

```diff
@@ -488,6 +702,170 @@ def get_class_ct(self, class_: str, ct_prop_name='rdfs:label'):
+                        for example: with identifier = 'short_label', classes = ['shortlabel1', 'shortlabel2']
+        :param ct_props: list of controlled terminology props to collect eg:
+                         ['label', 'Codelist Code', 'Order']
+                         defaults to ['rdfs:label']
+        :param identifier: string used when identifying classes
+        :return: Dictionary of classes and found controlled terminology for example:
+                 with classes = ['class1', 'class2'] and ct_props = ['label'] a typical result might be:
+                {'class1': [{'label': 'term1'}, {'label': 'term2'}], 'class2': [{{'label': 'term3'}}]}
+                Where 'class1' has two terms labeled 'term1' and 'term2' and 'class2' has one term labeled 'term3'
+        """
+
+        if ct_props is None:
+            ct_props = ['rdfs:label']
+        elif type(ct_props) == str:
+            ct_props = [ct_props]
+
+        prop_collection = f'["{ct_props[0]}", term.`{ct_props[0]}`]'
+        if len(ct_props) != 1:
+            for prop in ct_props[1:]:
+                prop_collection += f', ["{prop}", term.`{prop}`]'
+
+        q = f"""
+        UNWIND $classes as class
+        MATCH (c:Class)-[:HAS_CONTROLLED_TERM]->(term:Term)
+        WHERE c.`{identifier}` = class
+        WITH c, collect(apoc.map.fromPairs([{prop_collection}])) as terms
+        RETURN apoc.map.setKey({{}}, c.`{identifier}`, terms) as ct
+        """
+
+        params = {'classes': classes}
+        res = self.query(q, params)
+
+        data = {}
+        if res:
+            for term_dict in res:
+                # Flattened result {class.identifer:[[term.ct_props[0], ...], [term.ct_props[0], ...]]}
+                data.update(term_dict.get('ct'))
+        return data
+
+    def delete_ct(self, controlled_terminology: dict, term_props: list, identifier='label'):
+        """
+        Deletes class specific controlled terminology.
+        :param controlled_terminology: Dictionary of key: class identity - value: list of class specific term property
+                                       values to delete. For example:
+                                       with term_props = ['Codelist Code'] controlled terminology might be:
+                                      {'class1':[['code1'], ['code2']], class1':[['code2']]}
+                                      Note these property values must align with the keys defined in term_props!
+        :param term_props: List of property names to define controlled terminology term property values
+        :param identifier: string, property used to identify class
+        :return: neo4j result object.
+        """
+        # TODO: Resolving [:NEXT] rel when deleting CT
+
+        where_clause = f't.`{term_props[0]}` = term_props[0]'
+        for count, prop in enumerate(term_props[1:], start=1):
+            where_clause += f' AND t.`{prop}` = term_props[{count}]'
```

Comment:
a shorter code here and in similar places could be
```
WHERE apoc.map.submap(t, $ct_props) = apoc.map.submap(term_props, $ct_props)
```
However need to check if it impacts performance. It could that indexing in Neo4j works well on explicit `=` comparison but not with apoc.map.submap...